pynucastro/networks/rate_collection.py

```diff
@@ -881,7 +881,46 @@ def make_ap_pg_approx(self, intermediate_nuclei=None):
         # regenerate the links
         self._build_collection()
 
+    def make_nse_protons(self, A):
+        """for rates involving nuclei with mass number >= A, swap any
+        protons for NSE protons.  This will decouple these rates from
+        the proton captures at lower mass number, simplifying the
+        linear algebra.
+
+        """
+
+        # we want to update both the forward and reverse rates,
+        # so we are consistent
+        for rp in self.get_rate_pairs():
+
+            update = False
+            if rp.forward:
```

Comment:
probably better to do `if rp.forward is not None` for these

Reply:
I thought that was implicit here -- does it really make a difference?

Reply:
I just thought it's more explicit of what rp.forward/rp.reverse could be, just for clarity purposes.

Reply:
Okay, indeed pep-8 seems to agree

https://peps.python.org/pep-0008/#programming-recommendations

Reply:
updated